docs/root/version_history/current.rst

```diff
@@ -14,6 +14,7 @@ Minor Behavior Changes
 ----------------------
 *Changes that may cause incompatibilities for some users, but should not for most*
 
+* grpc_json_transcoder: filter now adheres to encoder and decoder buffer limits. Requests and responses that require buffering over the limits will be directly rejected.
```

Comment:
what happened previously?  Do these changes need to be runtime guarded per CONTRIBUTING.md?

Reply:
This is a behavior change. Previously the filter would accept and transcode messages that went over the buffer limit.

This filter is **not** marked as alpha in the [thread model](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/threat_model), so it should qualify for runtime guarding.

I was under the impression that security fixes are not breaking changes, but correct me if I'm wrong. I can add a guard feature that defaults to `true`. Can any other @envoyproxy/maintainers comment on this?

Reply:
my guess is the change will then cause problems for folks sending large responses, so IMO it's worth runtime gurading and maybe even adding a pointer to config to change if it causes errors.  Happy if you want to hold off until someone else chimes in.

Reply:
Thanks, makes sense. Added the runtime guard and documented the breaking change in the version history. Added a couple of unit tests with the runtime feature disabled.